system/ee/legacy/libraries/Filemanager.php

```diff
@@ -308,12 +308,17 @@ function get_image_dimensions($file_path)
 		{
 			$D = @getimagesize($file_path);
 
-			$image_size = array(
-				'height'	=> $D['1'],
-				'width'	=> $D['0']
+			if($D && is_array($D)) {
```

Comment:
Since you're making updates to this code it would be a good time to change that variable name from `$D` to something useful 😄

Reply:
I mean, what's more descriptive than `$D`?

One thing I am finding here, that I would appreciate help on, is when this returns false, it is breaking tests. It's almost as if `getimagesize` is failing, though it is a basic PHP function. File is found, path is valid, but getimagesize returns null. Any thoughts?